### PR TITLE
Fix init WebSocketService with ClusterData

### LINKED ISSUE #11233: Failed to retrieve Cluster data due to empty ConfigurationStoreServers
**Describe the bug**
Start pulsar proxy with WebSocketServiceEnabled=true in command line.

**Additional context**
```log
org.apache.pulsar.broker.PulsarServerException: Failed to retrieve Cluster data due to empty ConfigurationStoreServers
	at org.apache.pulsar.websocket.WebSocketService.retrieveClusterData(WebSocketService.java:225) ~[classes/:?]
	at org.apache.pulsar.websocket.WebSocketService.getPulsarClient(WebSocketService.java:164) ~[classes/:?]
	at org.apache.pulsar.websocket.ConsumerHandler.<init>(ConsumerHandler.java:103) ~[classes/:?]
```

### PR DESCRIPTION
**Pull request by Technoboy-**
Fixes #11233 .

### Motivation
1. Fix init WebSocketService with ClusterData.(before is null).
2. As ProxyServiceStarter is using in script or command line, no test case is covered. So it's better to add test case . But this class inits from config file, it's difficult to set the port or config due to the dynamic generation of port in the test case. So I adjusted the implementation of the class. 
3. Add test case for ProxyServiceStarter.

### CODE REVIEW

File: pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyServiceStarterTest.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    @Override
+    @BeforeClass
+    protected void setup() throws Exception {
+        internalSetup();
+        String[] args = new String[]{"-c", "./src/test/resources/proxy.conf"};
+        ProxyServiceStarter serviceStarter = new ProxyServiceStarter(args);
+        serviceStarter.getConfig().setBrokerServiceURL(pulsar.getBrokerServiceUrl());
+        serviceStarter.getConfig().setBrokerWebServiceURL(pulsar.getWebServiceAddress());
+        serviceStarter.getConfig().setServicePort(Optional.of(11000));
+        serviceStarter.getConfig().setWebSocketServiceEnabled(true);
```
Comment (codelipenghui) [inline pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyServiceStarterTest.java:61]: Seems we are setting the configurations through the Java API, do we need to load the `proxy.conf` file?

File: pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyServiceStarterTest.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        Future<Session> consumerSession = consumerWebSocketClient.connect(consumerSocket, URI.create(consumeUri));
+        consumerSession.get().getRemote().sendPing(ByteBuffer.wrap("ping".getBytes()));
+        producerSession.get().getRemote().sendString(ObjectMapperFactory.getThreadLocal().writeValueAsString(produceRequest));
+        assertTrue(consumerSocket.getResponse().contains("ping"));
+        ProducerMessage message = ObjectMapperFactory.getThreadLocal().readValue(consumerSocket.getResponse(), ProducerMessage.class);
+        assertEquals(new String(Base64.getDecoder().decode(message.getPayload())), "my payload");
+    }
+
+    @Test
+    public void testEnableWebSocketServerWithErrorClusterData() throws Exception {
```
Comment (codelipenghui) [inline pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyServiceStarterTest.java:126]: Why the test method ends with the ErroClusterData, looks we do not introduce error cluster settings here?

**Review by codelipenghui — COMMENTED** [review]
Great work! 

The change LGTM, just left some minor comments about the tests

File: pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyServiceStarterTest.java (hunk 2 above)
Comment (Technoboy-) [reply pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyServiceStarterTest.java:126]: Sorry, a mistake.

**Review by eolivelli — APPROVED** [review]

File: pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyServiceStarterTest.java (hunk 1 above)
Comment (Technoboy-) [reply pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyServiceStarterTest.java:60]: > Seems we are setting the configurations through the Java API, do we need to load the `proxy.conf` file?

ProxyServiceStarter load the config from command line "-c" and do this in the setup method.
But our test cases that start broker service listening on port 0, which would listen a port randomly. 
We can't specified the brokerServerUrl in the proxy.conf then. for we dont know the port in actual. So I adjust the implemetation of ProxyServiceStarter, let it expose ProxyConfiguration, and we can set configs to runtime ones.

Comment (codelipenghui) [reply pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyServiceStarterTest.java:60]: I see. Thanks for the explanation

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**